impl_core/src/main/java/io/opencensus/implcore/trace/export/SpanExporterImpl.java

```diff
@@ -187,27 +177,43 @@ public void run() {
               }
             } while (spans.isEmpty());
           }
-          spansCopy = new ArrayList<RecordEventsSpanImpl>(spans);
+          spansCopy = new ArrayList<>(spans);
           spans.clear();
         }
         // Execute the batch export outside the synchronized to not block all producers.
-        final List<SpanData> spanDataList = fromSpanImplToSpanData(spansCopy);
-        if (!spanDataList.isEmpty()) {
-          onBatchExport(spanDataList);
-        }
+        exportBatches(spansCopy);
       }
     }
 
-    void flush() {
-      List<RecordEventsSpanImpl> spansCopy;
+    private void flush() {
+      ArrayList<RecordEventsSpanImpl> spansCopy;
       synchronized (monitor) {
-        spansCopy = new ArrayList<RecordEventsSpanImpl>(spans);
+        spansCopy = new ArrayList<>(spans);
         spans.clear();
       }
+      // Execute the batch export outside the synchronized to not block all producers.
+      exportBatches(spansCopy);
+    }
 
-      final List<SpanData> spanDataList = fromSpanImplToSpanData(spansCopy);
+    private void exportBatches(ArrayList<RecordEventsSpanImpl> spanList) {
```

Comment:
It might be worth noting that spanList is assumed not to be shared across other threads.